tools/benchmark-cli/src/main/java/org/logstash/benchmark/cli/ui/UserInput.java

```diff
@@ -32,7 +32,7 @@
     public static final String TEST_CASE_PARAM = "testcase";
 
     public static final String TEST_CASE_HELP =
-        "Currently available test cases are 'baseline' and 'apache'.";
+        "Currently available test cases are 'baseline', 'apache' and 'custom'.";
```

Comment:
Maybe we should just remove this param and simply allow setting the new config param instead to simplify things a little?

Reply:
We could, do we not need to maintain backwards compatibility ?

Reply:
Nah, I don't think so :)

Reply:
Hi @original-brownbear, we need to support 3 types of configs / test cases - 

1. Baseline - This uses the logstash _generator_ plugin to generate random input data. No config file is needed for this case.
2. Apache - uses the sample apache config and a test data set downloaded from the provided url. Config file alone is not sufficient in this case, the data set is needed as well.
3. Custom - Need path to the custom config to run benchmarking against. 

So, for #1, we dont have the path to a config file to specify. For #2, we need to invoke a custom apache class that takes care of downloading the apache data set. 

I can combine all three of the test cases above into a single input flag - "config". _Config_ value would be "baseline", "apache" or path to a custom file. 

cc @jakelandis @jsvd

Reply:
Went through the code again.  Given the different handling required for the different test cases - baseline, apache and custom, I dont think it is a good idea to have a single config flag for all three, since the data type of the input will be different in each case. 

See @jakelandis comments in this [issue ]( #7784 ) - "Add in an option for --testcase=custom and expose a --conifg-dir option such that, assuming I can figure out how to feed data in, I can use this for any test case."